src/core/controllers/base.go

```diff
@@ -108,8 +108,47 @@ func (cc *CommonController) Login() {
 	cc.PopulateUserSession(*user)
 }
 
-// LogOut Habor UI
+// LogOut Harbor UI
 func (cc *CommonController) LogOut() {
+	// redirect for OIDC logout, excludes the admin user.
+	securityCtx, ok := security.FromContext(cc.Context())
+	if !ok {
+		log.Error("Failed to get security context")
+		cc.CustomAbort(http.StatusInternalServerError, "Internal error.")
+	}
+	principal := securityCtx.GetUsername()
+	if principal != "" {
+		if redirectForOIDC(cc.Ctx.Request.Context(), principal) {
+			u, err := user.Ctl.GetByName(cc.Context(), principal)
+			if err != nil {
+				log.Errorf("Error fetching user by name: %s, error: %v", principal, err)
+				cc.CustomAbort(http.StatusInternalServerError, "Internal error.")
+			}
+			if u == nil {
+				log.Errorf("User not found: %s, empty result", principal)
+				cc.CustomAbort(http.StatusInternalServerError, "Internal error.")
+			}
+			if u.UserID != 1 {
+				ep, err := config.ExtEndpoint()
+				if err != nil {
+					log.Errorf("Failed to get the external endpoint, error: %v", err)
+					cc.CustomAbort(http.StatusUnauthorized, "")
+				}
+				url := strings.TrimSuffix(ep, "/") + common.OIDCLoginoutPath
```

Comment:
I don't think you need to endpoint to redirect the user under the same hostname

Reply:
keep the redirection per the offline discussion, in the base handler, the render is set to return nil which is responsible for redirection. So, it needs to redirect to the OIDCLoginoutPath, and then to the IDP to logout user.